pkg/controller/direct/cloudbuild/workerpool_controller.go

```diff
@@ -198,19 +205,11 @@ func (a *Adapter) Create(ctx context.Context, u *unstructured.Unstructured) erro
 	if mapCtx.Err() != nil {
 		return mapCtx.Err()
 	}
-	resRef, err := NewResourceRef(created)
-	if err != nil {
-		return err
-	}
-	status.ExternalRef = resRef.GetExternalReference()
+	status.ExternalRef = asExternalRef(a.id)
```

Comment:
Yeah, I think this would read more fluently if asExternalRef was a method on the struct

Reply:
Gotcha. Updated.